### PR TITLE
Underwater Overlay Render Hook

### PR DESCRIPTION
**Pull request by shadowfacts**
Allows for items (such as goggles) that allow you to see properly underwater.

### CODE REVIEW

File: src/main/java/net/minecraftforge/client/event/RenderUnderwaterOverlayEvent.java [hunk 1]
```diff
@@ -0,0 +1,14 @@
+package net.minecraftforge.client.event;
+
+import net.minecraftforge.fml.common.eventhandler.Cancelable;
+import net.minecraftforge.fml.common.eventhandler.Event;
+
+/**
+ * This event is fired before the underwater overlay is rendered
+ * This event is {@link Cancelable}
+ * This event is fired from {@link net.minecraft.client.renderer.ItemRenderer#renderWaterOverlayTexture(float)}
```
Comment (williewillus) [inline src/main/java/net/minecraftforge/client/event/RenderUnderwaterOverlayEvent.java]: Describe what it does if canceled

Comment (shadowfacts) [inline src/main/java/net/minecraftforge/client/event/RenderUnderwaterOverlayEvent.java]: Fixed

File: src/main/java/net/minecraftforge/client/event/RenderUnderwaterOverlayEvent.java [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+/**
+ * This event is fired before the underwater overlay is rendered
+ * This event is {@link Cancelable}
+ * If canceled, the underwater overlay will not be rendered.
+ * This event is fired from {@link net.minecraft.client.renderer.ItemRenderer#renderWaterOverlayTexture(float)}
+ */
+@Cancelable
+public class RenderUnderwaterOverlayEvent extends Event
+{
```
Comment (LexManos) [inline src/main/java/net/minecraftforge/client/event/RenderUnderwaterOverlayEvent.java]: Really you don't think any information is useful in this event?

Comment (shadowfacts) [inline src/main/java/net/minecraftforge/client/event/RenderUnderwaterOverlayEvent.java]: I don't see anything useful available from where the event is fired that's not already available as a singleton.

**Comment by Leviathan143** [comment]
Doesn't this already exist as part of RenderBlockOverlayEvent?
https://github.com/MinecraftForge/MinecraftForge/blob/master/patches/minecraft/net/minecraft/client/renderer/ItemRenderer.java.patch#47

**Comment by shadowfacts** [comment]
/me facedesks

Oh god, I can't believe I missed that.

**Pull request closed without merging**